File: entry_point/perl_entry_point/entry_point.pl
#!/usr/bin/perl -w

use strict;
use warnings;

sub main {
  if( $#ARGV+1 != 1 ) {
    print STDERR "Usage: $0 <arg1>\n";
    exit(1);
  }
  print "$ARGV[0]\n";
  exit(0);
}

&main();

File: entry_point/ocaml_entry_point/entry_point.ml
let main () =
  if  Array.length Sys.argv <> 2 then
    begin
      Printf.printf "%s <arg1>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
      print_endline "begining of program"
    end

let _ = main ()
